### PR TITLE
fix submit buttons

### PR DESCRIPTION
**Pull request by xithan**
make disable_with a data attribute

### CODE REVIEW

File: card/mod/05_standard/set/abstract/attachment.rb [hunk 1]
```diff
@@ -117,7 +117,7 @@ def original_filename
 end
 
 def unfilled?
-  !attachment.present? && !save_preliminary_upload? && super
+  !attachment.present? && !save_preliminary_upload? && !subcards.present?
```
Comment (xithan) [inline card/mod/05_standard/set/abstract/attachment.rb]: change 1 of 2: consider attachment cards with content but without attachment as empty

File: card/mod/05_standard/set/all/rich_html/form.rb [hunk 2]
```diff
@@ -212,31 +233,52 @@ def process_relative_tags args
...
     form.radio_button method, tag_value, options
   end
 
+  def submit_button args={}
+    args.reverse_merge!(
+      situation: 'primary',
+      data: {}
+    )
+    text = args.delete(:text) || 'Submit'
+    args[:data][:disable_with] ||= args.delete(:disable_with) || 'Submitting'
```
Comment (xithan) [inline card/mod/05_standard/set/all/rich_html/form.rb]: change 2 of 2: make :disable_with a data argument

File: card/mod/01_core/set/all/type.rb [hunk 3]
```diff
@@ -18,6 +18,10 @@ def type_name
   type_card.try :name
 end
 
+def type_name_or_default
+  type_card.try(:name) || Card.quick_fetch(Card.default_type_id).name
```
Comment (ethn) [inline card/mod/01_core/set/all/type.rb]: fwiw, could do `type_name ||`

File: card/mod/05_standard/set/abstract/attachment.rb (hunk 1 above)
Comment (ethn) [inline card/mod/05_standard/set/abstract/attachment.rb]: makes sense.

**Pull request merged**